.github/workflows/tests.yml

```diff
@@ -24,12 +30,44 @@ jobs:
       with:
         node-version: '10.x'
 
-    - name: Install dependencies
+    - name: Install python setup dependencies
+      run: |
+        pip install --user --upgrade pip setuptools wheel
+
+    - name: Get pip cache dir
+      id: pip-cache
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install --upgrade pre-commit
-        pip install -e .
-        yarn
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache pip
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
```

Comment:
sure, can drop the cache.

Reply:
well if it's really useful then we should keep it, I just think we should be in the habit of justifying added complexity and maintenance burden, so I'm wondering what benefit it brings! (not out of skepticism, just because I do not know)

Reply:
simplified on 8e122bb1650a416cb5e90d3472700509c73fe02a etc

Reply:
Using the CI cache makes lighter use of community resources, e.g. pypi. It's not making a detectable _speed_ difference, however in this case.

To _really_ drive down CLI complexity, I can highly recommend `doit`... it's nice for multi-language projects this.